builder/dockerfile/parser/parser.go

```diff
@@ -204,6 +213,13 @@ func Parse(rwc io.Reader, d *Directive) (*Node, error) {
 		}
 
 		if child != nil {
+			if warning {
+				fmt.Fprintf(stderr, "[WARNING]: Empty lines detected in the following instruction:\n")
+				for _, line := range strings.Split(scannedLine, "\n") {
+					fmt.Fprintf(stderr, "[WARNING]:   %s\n", line)
+				}
+				fmt.Fprintf(stderr, "[WARNING]: Empty lines inside a RUN instruction is deprecated, and will be removed in Docker 1.16.\n")
```

Comment:
@yongtang per the discussion, perhaps we should remove this line (or at least remove the "version"). We discussed this in the maintainers meeting, and we may have to accept that this behaviour is "bad" (hence the warnings to motivate people to change it), but that removing would be too much of a backward breaking change, that we'd likely never be able to actually remove it.